### DISCUSSION TITLE
.env variables for Preview - vercel-cli

### DISCUSSION THREAD

**Discussion by adams-family** [vercel/community#2751] [Help]
### Summary

When deploying with vercel-cli, if an `.env.production` file is present in the root folder, it's content is used and exposed to the environment of the Vercel app.

Environment variables for Preview are mostly different from Production and my question is: how can we specify different environment variables for Production and for Preview?

I tried `.env.dev`, `.env.preview`, `.env.local`, etc... but nothing seems to work.

### Steps to Reproduce

```
$ vercel
```

**Comment by amyegan** [comment]
Hi @adams-family! The CLI looks in a special `.vercel` directory for [environment variables](https://vercel.com/guides/how-to-add-vercel-environment-variables) when building. You can use [`vercel pull`](https://vercel.com/docs/cli/pull#) with the `--environment` option to pull environment variables for a specific environment onto your computer. 

Some frameworks handle additional sources of environment variables, but I'm not sure what framework is used in your project. Next.js, for example, has an [Environment Variable Load Order](https://nextjs.org/docs/pages/building-your-application/configuring/environment-variables#environment-variable-load-order) that determines env var source priority.

**Comment by adams-family** [comment]
Hi @amyegan - perfect, this makes a lot of sense to me!

I'm using Vite (Vue) and React (create-react-app) frameworks - if I understand correctly then what files are used to read environment variables depend on the UI framework, not on Vercel itself. This actually clears a lot of confusion for me.

You also mentioned that the CLI looks in the `.vercel` directory for environment variables. The link to the documentation above shows options to specify different environment variables for different environment (dev, preview, prod) on the UI, but doesn't explain whether it's possible to have different sets of variables defined for different environments i the `.vercel` folder? Thanks!
